contracts/src/v0.8/ccip/pools/GHO/UpgradeableBurnMintTokenPool.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import {Initializable} from "solidity-utils/contracts/transparent-proxy/Initializable.sol";
+import {ITypeAndVersion} from "../../../shared/interfaces/ITypeAndVersion.sol";
+import {IBurnMintERC20} from "../../../shared/token/ERC20/IBurnMintERC20.sol";
+
+import {UpgradeableBurnMintTokenPoolAbstract} from "./UpgradeableBurnMintTokenPoolAbstract.sol";
+import {UpgradeableTokenPool} from "./UpgradeableTokenPool.sol";
+
+import {IRouter} from "../../interfaces/IRouter.sol";
+
+/// @title UpgradeableBurnMintTokenPool
+/// @author Aave Labs
+/// @notice Upgradeable version of Chainlink's CCIP BurnMintTokenPool
+/// @dev Contract adaptations:
+/// - Implementation of Initializable to allow upgrades
+/// - Move of allowlist and router definition to initialization stage
+/// - Inclusion of rate limit admin who may configure rate limits in addition to owner
+contract UpgradeableBurnMintTokenPool is Initializable, UpgradeableBurnMintTokenPoolAbstract, ITypeAndVersion {
+  string public constant override typeAndVersion = "BurnMintTokenPool 1.5.0";
+
+  /// @dev Constructor
+  /// @param token The bridgeable token that is managed by this pool.
+  /// @param rmnProxy The address of the arm proxy
+  /// @param allowlistEnabled True if pool is set to access-controlled mode, false otherwise
+  constructor(
+    address token,
+    address rmnProxy,
+    bool allowlistEnabled
+  ) UpgradeableTokenPool(IBurnMintERC20(token), rmnProxy, allowlistEnabled) {
+    _disableInitializers();
+  }
+
+  /// @dev Initializer
+  /// @dev The address passed as `owner_` must accept ownership after initialization.
+  /// @dev The `allowlist` is only effective if pool is set to access-controlled mode
+  /// @param owner_ The address of the owner
+  /// @param allowlist A set of addresses allowed to trigger lockOrBurn as original senders
+  /// @param router The address of the router
+  function initialize(address owner_, address[] memory allowlist, address router) public virtual initializer {
```

Comment:
yep normally i'd agree, it's actually done here because `owner` shadows a public method with the same name in the inherited `ConfirmedOwnerWithProposal` contract. can rename here since the shadow if a function with the same name, so should not cause any runtime issues; lemme know if you feel strongly about updating here, will do :) 
<img width="830" alt="image" src="https://github.com/user-attachments/assets/5bb59e12-baad-4672-8b83-8c13460e7096">